pkg/transform/kubernetes/populators_test.go

```diff
@@ -588,6 +588,53 @@ func TestPopulateEnvFrom(t *testing.T) {
 	}
 }
 
+func TestPopulateVolumes(t *testing.T) {
+	volumeClaims := []spec.VolumeClaim{{Name: "foo"}, {Name: "bar"}, {Name: "barfoo"}}
+	volumes := []api_v1.Volume{{Name: "foo"}}
+
+	// a volumeMount is defined but that is not there in volumeClaims
+	// neither it is in pod level volumes, so this should fail
+	failingContainers := []api_v1.Container{
+		{VolumeMounts: []api_v1.VolumeMount{{Name: "baz"}}},
+	}
+
+	if _, err := populateVolumes(failingContainers, volumeClaims, volumes); err == nil {
+		t.Errorf("should have failed but passed for volumeMount that" +
+			" does not exist.")
+	}
+
+	passingContainers := []api_v1.Container{
```

Comment:
Pasting from other PR's comment:
---
Please see the code from Kubernetes again, the tests are separated on passing and failing behavior.

Passing tests, where output is matched: [here](https://github.com/kubernetes/apimachinery/blob/17c1f5d09ee69b7521aa62b2c9d641cdc0dc74e9/pkg/api/resource/quantity_test.go?utf8=%E2%9C%93#L227)

Failing tests, where you will certainly get an error and you are not matching anything just checking if the function being tested errored out: [here](https://github.com/kubernetes/apimachinery/blob/17c1f5d09ee69b7521aa62b2c9d641cdc0dc74e9/pkg/api/resource/quantity_test.go?utf8=%E2%9C%93#L478)